### ISSUE TITLE
repro: a Greek letter in parameters name cause an error

### ISSUE THREAD

**Issue by mmikhasenko** [iterative/dvc#5357]
# Bug Report

## Description

`dvc repro --dry` throws the error
```
ERROR: failed to reproduce 'dvc.yaml': Parameters 'background_fits.mÎ©b_bgd_range' are missing from 'settings.json'.
```
clearly misreplacing the greek symbol `Ω`

The name of the parameter is correct in dvc.yaml
```
    - settings.json:
      - background_fits.mΩb_bgd_range
```

It is correct in settings.json:
```
       "background_fits": {
                              "mΩb_bgd_range": [
                                                 6.1,
                                                 6.4
                                               ]
                          },
```
Also fine in dvc.lock
```
      background_fits.mΩb_bgd_range:
      - 6.1
      - 6.4
```

### Environment information

**Output of `dvc version`:**

```console
$ dvc version
```

```
λ dvc version
DVC version: 1.11.10 (exe)
---------------------------------
Platform: Python 3.7.9 on Windows-10-10.0.19041-SP0
Supports: All remotes
Cache types: hardlink
Cache directory: NTFS on C:\
Caches: local
Remotes: None
Workspace directory: NTFS on C:\
Repo: dvc, git
```

**Comment by skshetry** [comment]
It could be because we are not opening the file as `utf-8` encoding here:
https://github.com/iterative/dvc/blob/08300d3bcf51f7bd9c8509924110416461b11b60/dvc/dvcfile.py#L134

**Comment by isidentical** [comment]
> It could be because we are not opening the file as utf-8 encoding here:

Yes, that is also what I concluded would be the reason (now trying to write a reproducer)

**Comment by mmikhasenko** [comment]
thanks, guys, for figuring it out!
Do you know if I have to wait for the release on Windows or can pull artifacts produced from master once it is merged?

**Comment by isidentical** [comment]
>  Do you know if I have to wait for the release on Windows or can pull artifacts produced from master once it is merged?

You can possibly install the dvc again with `pip install git+https://github.com/iterative/dvc.git`, or wait for the upcoming pre-release (in a couple of weeks, I presume)

**Comment by skshetry** [comment]
@isidentical, let's backport it then? 
cc @efiop

**Comment by efiop** [comment]
@skshetry Do we need to do other adjustments per https://github.com/iterative/dvc/pull/5363#discussion_r566809750 ?

**Comment by skshetry** [comment]
Just to clarify, the adjustment is unrelated to this PR, so we can release the backport without it.
